smac/smac_cli.py

```diff
@@ -83,12 +85,43 @@ def main_cli(self):
                     fn=traj_fn, cs=scen.cs)
                 initial_configs.append(trajectory[-1]["incumbent"])
 
+        # Restore state
+        stats = None
+        incumbent = None
+        if args_.restore_state:
+            root_logger.debug("Restoring state from %s...", args_.restore_state)
+            # Check for folder and files
+            rh_path = os.path.join(args_.restore_state, "runhistory.json")
+            stats_path = os.path.join(args_.restore_state, "stats.json")
+            traj_path = os.path.join(args_.restore_state, "traj_aclib2.json")
+            scen_path = os.path.join(args_.restore_state, "scenario.txt")
+            if not os.path.isdir(args_.restore_state):
+               raise FileNotFoundError("Could not find folder from which to restore.")
+            # Load runhistory and stats
+            rh = RunHistory(aggregate_func=None)
+            rh.load_json(rh_path, scen.cs)
+            root_logger.debug("Restored runhistory from %s", rh_path)
+            stats = Stats(scen)
+            stats.load(stats_path)
+            root_logger.debug("Restored stats from %s", stats_path)
+            trajectory = TrajLogger.read_traj_aclib_format(
+                fn=traj_path, cs=scen.cs)
+            incumbent = trajectory[-1]["incumbent"]
+            root_logger.debug("Restored incumbent %s from %s", incumbent, traj_path)
+            # Copy traj if output_dir is different
+            if scen.output_dir != Scenario(scen_path).output_dir:
```

Comment:
By leaving it open to the user to specify the scenario, the user can choose to use a different scenario-file (with extended limits or another output-path) to continue a restored scenario. When continuing in a new output-folder, everything gets written anew except for the trajectory (which is written continously), so its copied. I changed the `Scenario(scen_path).output_dir` to `InputReader().read_scenario_file(scen_path)['output_dir']` to avoid possible side-effects.
Do you think the procedure is correct in general?